ldpd/ldpd.c

```diff
@@ -140,6 +140,7 @@ static struct option longopts[] =
 	{ "ctl_socket",  required_argument, NULL, OPTION_CTLSOCK},
 	{ "user",        required_argument, NULL, 'u'},
 	{ "group",       required_argument, NULL, 'g'},
+	{ "instance",    required_argument, NULL, 'n'},
```

Comment:
Ok, but if a daemon has the --instance parameter one would expect that you can run multiple instances of it on the same namespace/VRF, which is not the case for ldpd. I think you're abusing the instance ID parameter for something it's not intended to do.

Also, how would the external label manager distinguish different instances of ospfd (SR)? With your current solution, you would add the constraint that ospfd instances running on different namespaces should use unique instances IDs. Not cool.

My suggestion would be to extend the label manager to identify the clients by their namespace in addition to their protocol and instance ID.